FerramAerospaceResearch.Base/UnityJobs/OcclusionJobs.cs

```diff
@@ -0,0 +1,381 @@
+            right[index] = new float3(tmp.x, tmp.y, tmp.z);
+        }
+    }
+
+    [BurstCompile]
+    public struct AngleBetween : IJobParallelFor
+    {
+        [ReadOnly] public float3 dir;
+        [ReadOnly] public NativeArray<quaternion> rotations;
+        [WriteOnly] public NativeArray<float> angles;
+
+        public void Execute(int index)
+        {
+            float3 rot_dir = math.mul(rotations[index], new float3(0, 0, 1));
+            angles[index] = math.degrees(math.acos(math.min(math.max(math.dot(dir, rot_dir), -1f), 1f)));
+        }
+    }
+
+    [BurstCompile]
+    public struct GeneralDistanceInfo : IJob
+    {
+        [ReadOnly] public NativeArray<float> distances;
+        [WriteOnly] public NativeArray<float3> output;
+
+        public void Execute()
+        {
+            float min = float.PositiveInfinity;
+            float max = 0;
+            float total = 0;
+            for (int i = 0; i < distances.Length; i++)
+            {
+                total += distances[i];
+                max = math.max(max, distances[i]);
+                min = math.min(min, distances[i]);
+            }
+            output[0] = new float3(min, max, total / distances.Length);
+        }
+    }
+
+    [BurstCompile]
+    public struct DotProductJob : IJobParallelFor
+    {
+        [ReadOnly] public NativeArray<float3> A;
+        [ReadOnly] public float3 B;
+        [WriteOnly] public NativeArray<float> result;
+
+        public void Execute(int index)
+        {
+            result[index] = math.dot(A[index], B);
+        }
+    }
+
+    [BurstCompile]
+    public struct SortedFilterAndMapByDistanceJob : IJob
+    {
+        [ReadOnly] public NativeArray<quaternion> quaternions;
+        [ReadOnly] public NativeArray<float> distances;
+        [ReadOnly] public NativeArray<float> cutoffDistance;
+        public NativeList<SphereDistanceInfo> sdiList;
+        [WriteOnly] public NativeList<quaternion> orderedQuaternions;
+
+        public void Execute()
+        {
+            for (int i = 0; i < quaternions.Length; i++)
+            {
+                if (distances[i] < cutoffDistance[0])
+                {
+                    SphereDistanceInfo sdi = new SphereDistanceInfo
+                    {
+                        q = quaternions[i],
+                        distance = distances[i]
+                    };
+                    sdiList.Add(sdi);
+                }
+            }
+            NativeSortExtension.Sort(sdiList);
+            for (int i = 0; i < sdiList.Length; i++)
+                orderedQuaternions.Add(sdiList[i].q);
+        }
+    }
+
+    [BurstCompile]
+    public struct MakeQuaternionIndexMapJob : IJobParallelFor
+    {
+        [ReadOnly] public NativeArray<quaternion> arr;
+        [WriteOnly] public NativeHashMap<quaternion, int>.ParallelWriter map;
+
+        public void Execute(int index)
+        {
+            map.TryAdd(arr[index], index);
```

Comment:
This will likely run faster in sequential mode since there's no extra work done besides writing

Reply:
Implemented in 53809f6.  Don't have the comparison from before, but some-random-runs is 0.026ms.